### ISSUE TITLE
Expose a DateTime type?

### ISSUE THREAD

**Issue by tailhook** [pyfisch/httpdate#1]
Hi I think it's beneficial to expose `DateTime` type, for two reasons:

1. It implements Display so can be written directly into output buffer without intermediate allocated String
2. It's also smaller than SystemTime

**Comment by pyfisch** [comment]
Hi, I want to hide `DateTime` as an implementation detail and only use the standard `SystemTime`. Avoiding the allocated string would be good. I could either provide a function like this:

```rust
pub fn format_http_date(f: &mut fmt::Formatter, d: SystemTime) -> fmt::Result {
    DateTime::from(d).fmt(f)
}
```

It avoids the intermediate string but it can only be called if you already have a formatter. The alternative would be a type ```pub struct HttpDate (SystemTime}``` and implement `FromStr` and `Display` for it. Do you know if the smaller size of `DateTime` actually makes a difference?

**Comment by tailhook** [comment]
Well, the tokio's structures are generally quite large. And sure, 16 bytes vs 8 bytes of timestamp does not matter here.

Still I'm not sure why you want a yet another wrapper. There is no need to expose any methods or internal structure of DateTime except FromStr, Display, From, Into and perhaps a PartialOrd. So both options will look the same from the point of view of a user.

**Comment by pyfisch** [comment]
> There is no need to expose any methods or internal structure of DateTime except FromStr, Display, From, Into and perhaps a PartialOrd. So both options will look the same from the point of view of a user.

Ah good idea. So `parse_http_date` would return an opaque `DateTime` and can be converted to a `SystemTime` if necessary. It can also be formatted with Display. Maybe `DateTime` is too generic would you agree to name it `HttpTime`?

**Comment by tailhook** [comment]
Yes `HttpTime` sounds good.

**Comment by pyfisch** [comment]
While editing the code I noticed that the crate is called `httpdate` but the type would be named `HttpTime`. Either I need to rename the crate and yank the old version or I should call it `HttpDate`.

**Comment by tailhook** [comment]
For that matter, I think `HttpDate` is equally meaningful as `HttpTime` :)

**Comment by pyfisch** [comment]
Sorry for the delay, can you please have a look at the publicdate branch? I will release it after your review.

**Comment by tailhook** [comment]
Looks good. Thanks!

**Comment by pyfisch** [comment]
Published as [v0.3.0](https://crates.io/crates/httpdate/0.3.0)
